File: emp_ui/lib/models/leave_request.dart
class LeaveRequestModel {
  final int? id;
  final int? employeeId;
  final String employeeCode;
  final String employeeName;
  final String leaveType;
  final String startDate;
  final String endDate;
  final String reason;
  final String leaveProposal;
  final String status;

  LeaveRequestModel({
    this.id,
    this.employeeId,
    required this.employeeCode,
    required this.employeeName,
    required this.leaveType,
    required this.startDate,
    required this.endDate,
    required this.reason,
    required this.leaveProposal,
    required this.status,
  });

  factory LeaveRequestModel.fromJson(Map<String, dynamic> json) {
    return LeaveRequestModel(
      id: json['id'],

      employeeId: json['employee'] != null ? json['employee']['id'] : null,
      employeeCode: json['employeeCode'] ?? '',
      employeeName: json['employeeName'] ?? 
                    (json['employee'] != null ? json['employee']['fullName'] : 'N/A'),
      leaveType: json['leaveType'] ?? '',
      startDate: json['startDate'] ?? '',
      endDate: json['endDate'] ?? '',
      reason: json['reason'] ?? '',
      leaveProposal: json['leaveProposal'] ?? '',
      status: json['status'] ?? 'PENDING',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      "id": id,

      "employee": employeeId != null ? {"id": employeeId} : null,
      "employeeCode": employeeCode,
      "employeeName": employeeName,
      "leaveType": leaveType,
      "startDate": startDate,
      "endDate": endDate,
      "reason": reason,
      "leaveProposal": leaveProposal,
      "status": status,
    };
  }
}

File: emp-api/src/main/java/com/example/entities/LeaveRequest.java
package com.example.entities;

import jakarta.persistence.*;
import java.time.LocalDate;

@Entity
@Table(name = "leave_request")
public class LeaveRequest {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String employeeCode;
    private String employeeName;

    private String leaveType;
    private LocalDate startDate;



    private LocalDate endDate;
    private String reason;
    private String leaveProposal;

    @Enumerated(EnumType.STRING)
    private LeaveStatus status; // PENDING, APPROVED, REJECTED

    // ===== Getters & Setters =====

    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }

    public String getEmployeeCode() { return employeeCode; }
    public void setEmployeeCode(String employeeCode) { this.employeeCode = employeeCode; }

    public String getEmployeeName() { return employeeName; }
    public void setEmployeeName(String employeeName) { this.employeeName = employeeName; }

    public String getLeaveType() { return leaveType; }
    public void setLeaveType(String leaveType) { this.leaveType = leaveType; }
    public LocalDate getStartDate() {
        return startDate;
    }

    public void setStartDate(LocalDate startDate) {
        this.startDate = startDate;
    }
    public LocalDate getEndDate() { return endDate; }
    public void setEndDate(LocalDate endDate) { this.endDate = endDate; }

    public String getReason() { return reason; }
    public void setReason(String reason) { this.reason = reason; }

    public String getLeaveProposal() { return leaveProposal; }
    public void setLeaveProposal(String leaveProposal) { this.leaveProposal = leaveProposal; }

    public LeaveStatus getStatus() { return status; }
    public void setStatus(LeaveStatus status) { this.status = status; }
}



